core/src/main/scala/org/graphframes/pattern/patterns.scala

```diff
@@ -45,8 +45,29 @@ private[graphframes] object PatternParser extends RegexParsers {
     "!" ~ edge ^^ { case _ ~ e =>
       Negation(e)
     }
```

Comment:
This is such a badly needed feature that... I can't believe it was solved by such few LOC :) Right on!

Reply:
@rjurney Thank you for reviewing my pull request!